atlp-rwanda/cabal-phantom-backend
# ft:added-routes-for-registering-drivers-and-operators

**What does this PR do?**
requests to merge ft-register-driversd-operators-9JpkBREJ with develop
 **Description of Task to be completed?**
Created CRUD routes for registering operators and drivers
Generate randomly passwords for registered users and hash them for security
Send them email of the unhashed password
**How should this be manually tested?**

-Use this **heroku** PR deployed app https://phantom-caba-ft-registe-m7fmun.herokuapp.com/phantom-docs/

- Clone this repo

- git checkout origin/ft-register-driversd-operators-9JpkBREJ

- install all packages

- make sure your db is connected with the help of .env.example file 
-log in your email using help of .env.example file 
N.B: two emails are required: t**he one to send password in auth function** and **the one to receive it**
       please use .env.example
- do npm run migrate

- go to your postman or  browse http://localhost:5000/phantom-docs/

1. GET for  all drivers and operators at: http://localhost:5000/api/v1/users/
2. POST at http://localhost:5000/api/v1/users/ to send an email to
3.PUT at http://localhost:5000/api/v1/users/(id)
4.DELETE at http://localhost:5000/api/v1/users/(id)
5.Use one of the users registered password to login


** What are the relevant pivotal tracker stories?**

[9JpkBREJ](https://trello.com/c/9JpkBREJ/10-register-drivers-operators
** Screenshots (if appropriate)**
<img width="1440" alt="Screen Shot 2020-10-09 at 09 29 19" src="https://user-images.githubusercontent.com/68019992/95555397-fff68580-0a11-11eb-834d-62f99ae5e27a.png">

<img width="1440" alt="Screen Shot 2020-10-14 at 07 49 04" src="https://user-images.githubusercontent.com/68019992/95948897-c9de4a80-0df1-11eb-9a1f-26a55971bfe6.png">
